fw/procfs.c

```diff
@@ -85,17 +106,42 @@ static int
 tfw_perfstat_seq_show(struct seq_file *seq, void *off)
 {
 #define SPRNE(m, e)	seq_printf(seq, m": %llu\n", e)
+#define SPRNED(m, e)	seq_printf(seq, m": %dms\n", e)
 #define SPRN(m, c)	seq_printf(seq, m": %llu\n", stat.c)
 
+	int i;
 	TfwPerfStat stat;
 	u64 serv_conn_active, serv_conn_sched;
 	SsStat *ss_stat = kmalloc(sizeof(SsStat) * num_online_cpus(),
 				  GFP_KERNEL);
+	unsigned int val[T_PSZ] = { 0 };
+	TfwPrcntlStats pstats = {.val = val};
+
 	if (!ss_stat)
 		T_WARN("Cannot allocate sync sockets statistics\n");
 
 	memset(&stat, 0, sizeof(stat));
+
+	if (health_stat_codes) {
+		stat.hm = kmalloc(tfw_hm_stats_size(health_stat_codes->ccnt),
```

Comment:
I am not particularly fond of constant allocation and freeing of small chunks of memory here. Perhaps, Tempesta's memory pool would be better suitable for this?